src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs

```diff
@@ -2433,7 +2431,7 @@ private void ExpandByABlock(int minBlockCharCount)
             // - We'd also prefer to make it at least at big as the current length (thus doubling capacity).
             //   - But this is only up to a maximum, so we stay in the small object heap, and never allocate
             //     really big chunks even if the string gets really big.
-            int newBlockLength = Math.Max(minBlockCharCount, Math.Min(Length, MaxChunkSize));
+            int newBlockLength = Math.Max(minBlockCharCount, Math.Min(Length * 2, MaxChunkSize));
```

Comment:
So this is tripling the capacity after expansion? That sounds very aggressive. Also, the comment above needs updating.